platform/util/platform-util/src/main/java/org/codice/ddf/platform/util/properties/PropertiesLoader.java

```diff
@@ -108,108 +122,140 @@ public Properties loadProperties(String propertiesFile) {
+    File propFile = new File(propertiesFile);
+    Properties properties = new Properties();
+
+    if (propFile.isAbsolute()) {
+      LOGGER.debug("propertiesFile {} is absolute", propertiesFile);
+      propFile = new File(propertiesFile);
+    } else {
+      String rootDirectory = System.getProperty("karaf.home");
+      if (rootDirectory != null && !rootDirectory.isEmpty()) {
+        propFile = new File(rootDirectory, propertiesFile);
+      } else {
+        rootDirectory = System.getProperty("ddf.home");
+        if (rootDirectory != null && !rootDirectory.isEmpty()) {
+          propFile = new File(rootDirectory, propertiesFile);
         } else {
-          String rootDirectory = System.getProperty("karaf.home");
-          if (rootDirectory != null && !rootDirectory.isEmpty()) {
-            propFile = new File(rootDirectory, propertiesFile);
-          } else {
-            rootDirectory = System.getProperty("ddf.home");
-            if (rootDirectory != null && !rootDirectory.isEmpty()) {
-              propFile = new File(rootDirectory, propertiesFile);
-            } else {
-              propFile = new File(propertiesFile);
-            }
-          }
-        }
-        properties = new Properties();
-
-        try (InputStreamReader reader =
-            new InputStreamReader(new FileInputStream(propFile), StandardCharsets.UTF_8)) {
-          properties.load(reader);
-        } catch (FileNotFoundException e) {
-          error = true;
-          LOGGER.debug("Could not find properties file: {}", propFile.getAbsolutePath(), e);
-        } catch (IOException e) {
-          error = true;
-          LOGGER.debug("Error reading properties file: {}", propFile.getAbsolutePath(), e);
-        }
-      }
-      if (error || properties.isEmpty()) {
-        LOGGER.debug("Attempting to load properties as a resource: {}", propertiesFile);
-        InputStream ins = PropertiesLoader.class.getResourceAsStream(propertiesFile);
-        if (ins != null) {
-          try {
-            properties.load(ins);
-            ins.close();
-          } catch (IOException e) {
-            LOGGER.debug("Unable to load properties: {}", propertiesFile, e);
-          } finally {
-            IOUtils.closeQuietly(ins);
-          }
+          propFile = new File(propertiesFile);
         }
       }
+    }
 
-      // replace any ${prop} with system properties
-      Properties filtered = new Properties();
-      for (Map.Entry<?, ?> entry : properties.entrySet()) {
-        filtered.put(
-            StrSubstitutor.replaceSystemProperties(entry.getKey()),
-            StrSubstitutor.replaceSystemProperties(entry.getValue()));
-      }
-      properties = filtered;
-
-    } else {
-      LOGGER.debug("Properties file must not be null.");
+    try (InputStreamReader reader =
+        new InputStreamReader(new FileInputStream(propFile), StandardCharsets.UTF_8)) {
+      properties.load(reader);
+    } catch (FileNotFoundException e) {
```

Comment:
✏️ We can't _not_ catch this checked exception as it is thrown from the FIS ctor, but we can eliminate the processing hit of it _actually_ occurring by confirming that `propFile` exists prior to entering this try/catch block.